src/components/ModalDialog/ModalDialog.vue

```diff
@@ -0,0 +1,282 @@
+            'unnnic-modal-dialog__container__actions--divider',
+        ]"
+      >
+        <UnnnicButton
+          type="tertiary"
+          :text="secondaryButtonText || i18n('cancel')"
+          @click.stop="
+            secondaryButtonText ? $emit('secondaryButtonClick') : close()
+          "
+        />
+        <UnnnicButton
+          :type="
+            primaryButtonType || primaryButtonTypeMapper[type] || 'primary'
+          "
+          :text="primaryButtonText"
+          @click.stop="$emit('primaryButtonClick')"
+        />
+      </section>
+    </section>
+  </section>
+</template>
+
+<script>
+import UnnnicIcon from '../Icon.vue';
+import UnnnicButton from '../Button/Button.vue';
+import UnnnicI18n from '../../mixins/i18n';
+
+export default {
+  name: 'UnnnicModalDialog',
+  components: {
+    UnnnicIcon,
+    UnnnicButton,
+  },
+  mixins: [UnnnicI18n],
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+    persistent: {
+      type: Boolean,
+      default: false,
+    },
+    type: {
+      type: String,
+      default: '',
+      validate(type) {
+        return ['success', 'warning', 'attention'].includes(type);
+      },
+    },
+    size: {
+      type: String,
+      default: 'md',
+      validate(size) {
+        return ['sm', 'md', 'lg'].includes(size);
+      },
+    },
+    title: {
+      type: String,
+      default: 'title',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    iconScheme: {
+      type: String,
+      default: '',
+    },
+    showCloseIcon: {
+      type: Boolean,
+      default: false,
+    },
+    primaryButtonText: {
+      type: String,
+      default: '',
+    },
+    secondaryButtonText: {
+      type: String,
+      default: '',
+    },
+    primaryButtonType: {
+      type: String,
+      default: '',
+    },
+    showActionsDivider: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  emits: ['primaryButtonClick', 'secondaryButtonClick', 'update:modelValue'],
+
+  data() {
+    return {
+      defaultTranslations: {
+        cancel: {
+          'pt-br': 'Cancelar',
+          en: 'Cancel',
+          es: 'Cancelar',
+        },
+      },
+      iconsMapper: {
+        success: { icon: 'check_circle', scheme: 'aux-green-500' },
+        warning: { icon: 'warning', scheme: 'aux-red-500' },
+        attention: { icon: 'error', scheme: 'aux-yellow-500' },
+      },
+      primaryButtonTypeMapper: {
+        success: 'primary',
+        warning: 'warning',
+        attention: 'attention',
+      },
+    };
+  },
+  watch: {
+    modelValue(value) {
+      if (value) document.body.style.overflow = 'hidden';
+      else document.body.style.overflow = '';
+    },
+  },
+  methods: {
+    close() {
+      this.$emit('update:modelValue', false);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@import '../../assets/scss/unnnic.scss';
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
```

Comment:
This code will not generate the same problem in the children's elements that have been corrected in [this Drawer PR](https://github.com/weni-ai/unnnic/pull/488/files#diff-196de012efad46f52969a72e5c3433754d9c025eb6762bd5f9abf7a68db68aa0L218-L222)?